[next] FS create folder before writing in it

WebReflection opened the pull request:
## Description

See https://github.com/pyscript/pyscript/issues/1580

## Changes

  * apply suggested changes to avoid analyzePath to throw
  * drop irrelevant unit tests around it (requires integration)

## Checklist

<!-- Note: Only user-facing changes require a changelog entry. Internal-only API changes do not require a changelog entry. Changes in documentation do not require a changelog entry. -->

-   [x] All tests pass locally
-   [ ] I have updated `docs/changelog.md`
-   [ ] I have created documentation for this(if applicable)

JeffersGlass commented:
Suspect this wants to target `next` and not `main` yes? Also it looks like #1581 was submitted, which is very similar.

WebReflection commented:
@JeffersGlass yes, the default to *master* instead of *next* is making my days lovely ... but also this one, as MR, doesn't ruin code coverage so I think this should be considered for now.

e-nikolov commented:
Looks like next to its FS API, Emscripten also has [PATH and PATH_FS](https://github.com/emscripten-core/emscripten/blob/main/src/library_path.js) that include things like dirname(), basename() and resolve(). Would it make sense to use them?

hoodmane commented on pyscript.core/esm/interpreter/_utils.js:
```diff
@@ -40,11 +39,12 @@ export const writeFile = (FS, path, buffer) => {
 // This is instead a fallback for Lua or others
 export const writeFileShim = (FS, path, buffer) => {
-    path = resolve(FS, path);
     mkdirTree(FS, dirname(path));
```
> why pyodide doesn't just allow access to `FS_PATH`

Haven't gotten around to it. I'm thinking about merging `FS_PATH` and `PATH` and exposing them as `pyodide.PATH` or `pyodide.FS.path` or something like that. But in the meantime, `pyodide._module.FS_PATH` is not likely to go anywhere.

WebReflection replied:
For now I am using `._module` as both Pyodide and MicroPython have some hidden stuff in it (in different places) but I am currently happy about the state of this MR.

WebReflection commented on pyscript.core/esm/interpreter/_utils.js:
```diff
@@ -30,7 +30,6 @@ export const stdio = (init) => {
-/* c8 ignore stop */
 
 // This should be the only helper needed for all Emscripten based FS exports
 export const writeFile = (FS, path, buffer) => {
```
Done

WebReflection commented:
@e-nikolov I think I have addressed all concerns and now it's fully based on native functionality.

One notable difference between pyodide and MicroPython is that Pyodide consider its root the `/pyodide/` folder while MicroPython doesn't have a special root folder.

This is, however, aligned with what previous PyScript had too so I think we should be good.

All manual tests passed without issues.

WebReflection commented:
Added an issue to *wasmoon* project: https://github.com/ceifa/wasmoon/issues/83

I'd love to fully get rid of the current ugly, quick, and inevitably dirty, code around that interpreter too.
